### DISCUSSION TITLE
How to pass -Wl,--start-group -lc_nano -lgcc -Wl,--end-group

### DISCUSSION THREAD

**Discussion by laomaolaile** [eclipse-embed-cdt/eclipse-plugins#591] [Debug Plug-ins Q&A]
In some projects, I would like my Libraries to be passed on as a group. How can I achieve this?

For example, in the following example, I actually want to pass -Wl,--start-group -lstdc++ -lc_nano -lgcc -Wl,--end-group

![image](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/19726057/fc945501-4c6b-466f-b40a-edadb9953001)

**Comment by TommyMurphyTM1234** [comment]
Isn't that question addressed here already?

* https://github.com/eclipse-embed-cdt/eclipse-plugins/issues/590

**Reply by laomaolaile** [reply]
I don't think it's the implementation I wanted, I would like to ask if there is a better way to achieve it

**Reply by TommyMurphyTM1234** [reply]
I don't understand - what exactly is the problem with the approach suggested by @ilg-ul here?

* https://github.com/eclipse-embed-cdt/eclipse-plugins/issues/590#issuecomment-1764096780

**Comment by laomaolaile** [comment]
Hi @TommyMurphyTM1234 ，

I want achive the following result, as screenshot below:

![image](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/19726057/e2b0557a-963a-423e-82f6-972350a27b7e)


As ld doc here https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_mono/ld.html, I want to resolve undefined symbol issue.

~~~
--start-group archives --end-group
The archives should be a list of archive files. They may be either explicit file names, or `-l' options. The specified archives are searched repeatedly until no new undefined references are created. Normally, an archive is searched only once in the order that it is specified on the command line. If a symbol in that archive is needed to resolve an undefined symbol referred to by an object in an archive that appears later on the command line, the linker would not be able to resolve that reference. By grouping the archives, they all be searched repeatedly until all possible references are resolved. Using this option has a significant performance cost. It is best to use it only when there are unavoidable circular references between two or more archives.
~~~

If no start group and end group to wrap the linked libary, it will show undefined symbols error during link, I am not using --specs gcc option with -nodefault-libs, and I want to control the libraries linked by me.

![image](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/19726057/368e124e-8373-4a8b-811c-2f487da6c304)

**Reply by TommyMurphyTM1234** [reply]
And @ilg-ul has already said:

> You can add additional linker options, but I'm not sure you can use variables (like LIBS), so you probably have to list the libraries explicitly.

* https://github.com/eclipse-embed-cdt/eclipse-plugins/issues/590#issuecomment-1764096780

**Comment by laomaolaile** [comment]
Hi @ilg-ul, the previous issue #590 is created by my collegue, and I provided more detailed information, could you take a look at it.
